tests/integrations/test_diffusion.py

```
@@ -0,0 +1,271 @@
+            mock_model, input_ids
+        )
+
+        # Loss should be zero when no tokens are masked
+        assert loss.item() == 0.0
+        assert loss.requires_grad
+
+    def test_cache_special_token_ids(self, diffusion_trainer_instance):
+        """Test caching of special token IDs."""
+        # Should cache BOS, EOS, PAD tokens
+        expected_tokens = {0, 1, 2}  # pad, bos, eos
+        assert diffusion_trainer_instance._special_token_ids == expected_tokens
```

Ruff (0.12.2)

Use of `assert` detected

(S101)